crates/transpiler/src/passes/vf2/vf2_layout.rs

```diff
@@ -51,21 +50,38 @@ impl EdgeList {
 
 create_exception!(qiskit, MultiQEncountered, pyo3::exceptions::PyException);
 
-fn build_average_error_map(target: &Target) -> ErrorMap {
-    let qargs_count = target.qargs().unwrap().count();
-    let mut error_map = ErrorMap::new(Some(qargs_count));
-    for qargs in target.qargs().unwrap() {
-        let mut qarg_error: f64 = 0.;
-        let mut count: usize = 0;
-        for op in target.operation_names_for_qargs(qargs).unwrap() {
-            if let Some(error) = target.get_error(op, qargs) {
-                count += 1;
-                qarg_error += error;
-            }
+/// Build an average error map for a given target.
+///
+/// Returns `None` if there is a global 2q operation to avoid attempting to construct a meaningless
+/// all-to-all connectivity graph.
+fn build_average_error_map(target: &Target) -> Option<ErrorMap> {
```

Comment:
This function looks quite a bit longer from the one before the change: are we building a more precise error map now?

Reply:
I think the high-degree thing is copied over from the old scoring heuristics - I have a vague memory of there being a test that specifically looked for error-rate ties to be broken by the node degree.